metadata-ingestion/src/datahub/ingestion/source/iceberg/iceberg.py

```diff
@@ -181,26 +197,20 @@ def _create_iceberg_workunit(
             aspects=[Status(removed=False)],
         )
 
-        custom_properties: Dict = dict(table.properties())
-        custom_properties["location"] = table.location()
-        try:
-            if isinstance(table, BaseTable) and table.current_snapshot():
-                custom_properties["snapshot-id"] = str(
-                    table.current_snapshot().snapshot_id
-                )
-                custom_properties[
-                    "manifest-list"
-                ] = table.current_snapshot().manifest_location
-        except KeyError:
-            # The above API is not well implemented, and can throw KeyError when there is no data.
-            pass
+        # Dataset properties aspect.
+        custom_properties = dict(table.metadata.properties)
+        custom_properties["location"] = table.metadata.location
+        if table.current_snapshot():
```

Comment:
We can leverage the walrus operator here to avoid calling `current_snapshot()` multiple times
```suggestion
        if snapshot := table.current_snapshot():
```

Reply:
Great suggestion, but unfortunately Datahub ingestion supports Python 3.7+ 😢 
https://github.com/datahub-project/datahub/blob/master/metadata-ingestion/setup.py#L602-L610